api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/PReLUTest.kt

```diff
@@ -5,111 +5,48 @@
 
 package org.jetbrains.kotlinx.dl.api.core.layer
 
-import org.jetbrains.kotlinx.dl.api.core.KGraph
-import org.jetbrains.kotlinx.dl.api.core.activation.EPS
 import org.jetbrains.kotlinx.dl.api.core.initializer.Constant
 import org.jetbrains.kotlinx.dl.api.core.initializer.RandomUniform
 import org.jetbrains.kotlinx.dl.api.core.layer.activation.PReLU
-import org.jetbrains.kotlinx.dl.api.core.shape.shapeFromDims
-import org.jetbrains.kotlinx.dl.api.core.shape.shapeOperand
-import org.junit.jupiter.api.Assertions
+import org.jetbrains.kotlinx.dl.api.core.shape.*
 import org.junit.jupiter.api.Test
-import org.tensorflow.Graph
-import org.tensorflow.Session
+import org.tensorflow.EagerSession
 import org.tensorflow.Shape
 import org.tensorflow.op.Ops
 
-class PReLUTest : ActivationLayerTest() {
+class PReLUTest : LayerTest() {
+
+    private val input = arrayOf(
+        floatArrayOf(-3.0f, -1.0f, 0.0f, 2.0f),
+        floatArrayOf(2.5f, 1.0f, -5.0f, 4.2f),
+    )
+
+    private val inputShape = input.shape.toLongArray()
+
     @Test
     fun default() {
-        val input = arrayOf(
-            floatArrayOf(-3.0f, -1.0f, 0.0f, 2.0f),
-            floatArrayOf(2.5f, 1.0f, -5.0f, 4.2f),
-        )
         val expected = arrayOf(
             floatArrayOf(0.0f, 0.0f, 0.0f, 2.0f),
             floatArrayOf(2.5f, 1.0f, 0.0f, 4.2f),
         )
         val layer = PReLU()
 
-        Graph().use { graph ->
-            Session(graph).use { session ->
-                val tf = Ops.create(graph)
-                KGraph(graph.toGraphDef()).use { kGraph ->
-                    val inputShape = Shape.make(input.size.toLong(), input[0].size.toLong())
-                    layer.build(tf, kGraph, inputShape)
-
-                    val inputOp = tf.constant(input)
-                    val isTraining = tf.constant(false)
-                    val numberOfLosses = tf.constant(1.0f)
-                    val output = layer.forward(tf, inputOp, isTraining, numberOfLosses).asOutput()
-                    kGraph.initializeGraphVariables(session)
-                    val outputTensor = session.runner().fetch(output).run().first()
+        // NOTE: It seems testing layers having trainable weights in Eager mode is not working.
+        //assertLayerOutputIsCorrect(layer, input, expected)
```

Comment:
@zaleslaw I remember you once mentioned that the Eager mode implementation in low-level API is buggy. Just to reconfirm that's actually the case? Because it really throws an error...

Reply:
eager mode in 1.15 runtime was limited and for complex things as layers I suggest to use GraphMode